src/misc/utils.ts

```diff
@@ -0,0 +1,9 @@
+export function forAll<T>(arr: T[] | Set<T>, cb: (arg0: T) => boolean): boolean {
```

Comment:
```suggestion
export function forAll<T>(iter: Iterable<T>, cb: (elem: T) => boolean): boolean {
```
I guess `Iterable` is a better candidate here.

Also, `forAll` can actualy be replaced with [`Array.some()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) (but not in case of `Set`s).

Maybe we can play around with `Array.prototype.some.call(iter, elem)` ([link](https://stackoverflow.com/a/56094977/2269749)) ?

Reply:
`some` seems to replace `forAny` not `forAll`?

Reply:
Fixed the type of arr to Iterable in ff5fa34